Misc/NEWS.d/next/Library/2025-04-06-21-17-14.gh-issue-132064.ktPwDM.rst

```diff
@@ -0,0 +1,4 @@
+:func:`annotationlib.get_annotations` now uses the ``__annotate__``
+attribute if it is present, even if ``__annotations__`` is not present.
+Additionally, the function now raises a :py:exc:`TypeError` if it is passed
+an object that does not have any annotatins.
```

Comment:
I'm _slightly_ worried that this might break some people's code unexpectedly if they were relying on the old behaviour, even though the old behaviour was the opposite to what the docs promised?

Reply:
The old behavior was only in the alphas, so I'm not concerned.

```
$ python3.13 -c 'from inspect import get_annotations; get_annotations(None)'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
    from inspect import get_annotations; get_annotations(None)
                                         ~~~~~~~~~~~~~~~^^^^^^
  File "/opt/homebrew/Cellar/python@3.13/3.13.1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/inspect.py", line 254, in get_annotations
    raise TypeError(f"{obj!r} is not a module, class, or callable.")
TypeError: None is not a module, class, or callable.
```

Reply:
Ah okay, I misunderstood and thought we were changing the behaviour from what `inspect.get_annotations` did on previous versions. Thanks!